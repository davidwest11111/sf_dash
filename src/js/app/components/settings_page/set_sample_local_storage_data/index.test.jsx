import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import LocalStorage, { BUS_STOP_ROUTE_TAGS_KEY } from 'app/utils/local_storage';

import SetSampleLocalStorageData from './index';


describe('outputs the expected tree when', () => {
  let wrapper;

  test('(default)', () => {
    wrapper = shallow((
      <SetSampleLocalStorageData localStorageKey={BUS_STOP_ROUTE_TAGS_KEY} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow((
      <SetSampleLocalStorageData localStorageKey={BUS_STOP_ROUTE_TAGS_KEY} />
    ));
    instance = wrapper.instance();
  });

  test('setSampleData', () => {
    const spy = jest.spyOn(LocalStorage, 'setSampleData');

    instance.setSampleData();

    expect(spy).toHaveBeenCalledWith(BUS_STOP_ROUTE_TAGS_KEY);
  });
});
