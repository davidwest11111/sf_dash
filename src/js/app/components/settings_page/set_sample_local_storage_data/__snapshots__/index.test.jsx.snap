// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<div
  className=""
>
  <p>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Set Sample LocalStorage Data
    </Button>
  </p>
  <p
    className="small"
  >
    Replace stored data with sample data.
  </p>
</div>
`;
